osbs/utils.py

```diff
@@ -321,6 +326,10 @@ def make_name_from_git(repo, branch, limit=53, separator='-', hash_size=5):
     shaval = sha256(full.encode('utf-8')).hexdigest()
     hash_str = shaval[:hash_size]
 
+    # Sanitize repo and branch names
+    repo = ''.join(filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(repo)))
+    branch = ''.join(filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(branch)))
```

Comment:
That's what I thought! But python3 has its own mind...

In python 3, `filter` always returns a filter generator, which can only be converted to `str` via `str.join`. And in python 2, `filter` will either return a `list`, or a `str`, depending on the `typeof` input. To keep things working for both python2 and python3, convert the input to `list` to force `filter` in python2 to return a `list`. This way, both the `list` and `filter generator` can both work with `str.join`.